src/main/java/io/jenkins/plugins/pipelinegraphview/utils/PipelineNodeUtil.java

```diff
@@ -47,11 +50,43 @@ public static String getDisplayName(@NonNull FlowNode node) {
                 : name;
     }
 
+    public static boolean isStep(FlowNode node) {
+        if (node != null) {
+            if (node instanceof AtomNode) {
+                return true;
+            }
+            if (node instanceof StepStartNode) {
+                StepStartNode stepStartNode = (StepStartNode) node;
+                boolean takesImplicitBlockArgument = false;
+                StepDescriptor sd = stepStartNode.getDescriptor();
+                if (sd != null) {
+                    takesImplicitBlockArgument = sd.takesImplicitBlockArgument();
+                }
+                return !isStage(node)
+                        && !isParallelBranch(node)
+                        && stepStartNode.isBody()
+                        && !takesImplicitBlockArgument;
+            }
+        }
+        return false;
+    }
+
     public static boolean isStage(FlowNode node) {
-        return node != null
-                && ((node.getAction(StageAction.class) != null)
-                        || (node.getAction(LabelAction.class) != null
-                                && node.getAction(ThreadNameAction.class) == null));
```

Comment:
The previous logic would include some steps that generate `StepStartNode` nodes in the pipeline graph/tree and incorrectly categorize them as stages.